test/unit-test/cellular_common_api_utest.c

```diff
@@ -0,0 +1,921 @@
+
+/**
+ * @brief Test that null handler case for Cellular_CommonInit.
+ */
+void test_Cellular_CommonInit_Null_Handler( void )
+{
+    CellularError_t cellularStatus = CELLULAR_SUCCESS;
+
+    _Cellular_LibInit_IgnoreAndReturn( CELLULAR_INVALID_HANDLE );
+
+    cellularStatus = Cellular_CommonInit( NULL, &cellularCommInterface );
+
+    TEST_ASSERT_EQUAL( CELLULAR_INVALID_HANDLE, cellularStatus );
+}
+
+/**
+ * @brief Test that happy path case for Cellular_CommonInit.
+ */
+void test_Cellular_CommonInit_Happy_Path( void )
+{
+    CellularError_t cellularStatus = CELLULAR_SUCCESS;
+    struct CellularContext handler;
+
+    _Cellular_LibInit_IgnoreAndReturn( CELLULAR_SUCCESS );
+
+    Cellular_ModuleInit_IgnoreAndReturn( CELLULAR_SUCCESS );
+
+    Cellular_ModuleEnableUE_IgnoreAndReturn( CELLULAR_SUCCESS );
+
+    Cellular_ModuleEnableUrc_IgnoreAndReturn( CELLULAR_SUCCESS );
+
+    cellularStatus = Cellular_CommonInit( ( CellularHandle_t * ) &handler, &cellularCommInterface );
+
+    TEST_ASSERT_EQUAL( CELLULAR_SUCCESS, cellularStatus );
+}
+
+/**
+ * @brief Test that null handler case for Cellular_CommonCleanup.
+ */
+void test_Cellular_CommonCleanup_Null_Handler( void )
+{
+    CellularError_t cellularStatus = CELLULAR_SUCCESS;
+
+    _Cellular_LibInit_IgnoreAndReturn( CELLULAR_INVALID_HANDLE );
+
+    cellularStatus = Cellular_CommonCleanup( NULL );
+
+    TEST_ASSERT_EQUAL( CELLULAR_INVALID_HANDLE, cellularStatus );
+}
+
+/**
+ * @brief Test that happy path case for Cellular_CommonCleanup.
+ */
+void test_Cellular_CommonCleanup_Happy_Path( void )
+{
+    CellularError_t cellularStatus = CELLULAR_SUCCESS;
+    CellularHandle_t handler;
+
+    Cellular_ModuleCleanUp_IgnoreAndReturn( CELLULAR_SUCCESS );
+    _Cellular_LibCleanup_IgnoreAndReturn( CELLULAR_SUCCESS );
+    cellularStatus = Cellular_CommonCleanup( handler );
+
+    TEST_ASSERT_EQUAL( CELLULAR_SUCCESS, cellularStatus );
+}
+
+/**
+ * @brief Test that null handler case for Cellular_CommonRegisterUrcNetworkRegistrationEventCallback.
+ */
+void test_Cellular_CommonRegisterUrcNetworkRegistrationEventCallback_Null_Handler( void )
+{
+    CellularError_t cellularStatus = CELLULAR_SUCCESS;
+
+    _Cellular_CheckLibraryStatus_IgnoreAndReturn( CELLULAR_INVALID_HANDLE );
+
+    cellularStatus = Cellular_CommonRegisterUrcNetworkRegistrationEventCallback( NULL, NULL, NULL );
+
+    TEST_ASSERT_EQUAL( CELLULAR_INVALID_HANDLE, cellularStatus );
+}
+
+/**
+ * @brief Test that happy path case for Cellular_CommonRegisterUrcNetworkRegistrationEventCallback.
+ */
+void test_Cellular_CommonRegisterUrcNetworkRegistrationEventCallback_Happy_Path( void )
+{
+    CellularError_t cellularStatus = CELLULAR_SUCCESS;
+    CellularContext_t context;
+
+    _Cellular_CheckLibraryStatus_IgnoreAndReturn( CELLULAR_SUCCESS );
+    cellularStatus = Cellular_CommonRegisterUrcNetworkRegistrationEventCallback( ( CellularHandle_t ) &context, NULL, NULL );
+
+    TEST_ASSERT_EQUAL( CELLULAR_SUCCESS, cellularStatus );

```

Comment:
Is there a way to assert that the call back was actually stored for all these callback registration functions? 

In other words, can we detect if the following 2 lines are commented:

```c
    CellularContext_t * pContext = ( CellularContext_t * ) cellularHandle;
    CellularError_t cellularStatus = CELLULAR_SUCCESS;

    /* pContext is checked in _Cellular_CheckLibraryStatus function. */
    cellularStatus = _Cellular_CheckLibraryStatus( pContext );

    if( cellularStatus != CELLULAR_SUCCESS )
    {
        CellularLogDebug( "_Cellular_CheckLibraryStatus failed" );
    }
    else
    {
        /* pContext->cbEvents.networkRegistrationCallback = networkRegistrationCallback; */
        /* pContext->cbEvents.pNetworkRegistrationCallbackContext = pCallbackContext; */
    }

    return cellularStatus;
```

Reply:
I think I've done this for other APIs: https://github.com/andysun2015/Lab-Project-FreeRTOS-Cellular-Library/blob/05327159b70598096def9858db2135aca254ccb4/test/unit-test/cellular_common_api_utest.c#L804

I will add these mechanism for this API.